### PR TITLE
[SYCL] Small fixes and adjusments around sycl exceptions. 

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

File: sycl/include/CL/sycl/handler.hpp [hunk 1]
```diff
@@ -382,8 +382,11 @@ class handler {
           std::move(MSharedPtrStorage), std::move(MRequirements),
           std::move(MEvents)));
       break;
+    case detail::CG::NONE:
+      throw runtime_error("Command group submitted without a kernel or a "
+                          "explicit memory operation.");
     default:
-      throw runtime_error("Unhandled type of command group");
+      throw runtime_error("Unhanded type of command group");
```
Comment (jcranmer-intel) [inline sycl/include/CL/sycl/handler.hpp]: "Unhandled" is more correct than "Unhanded" here.

Comment (romanovvlad) [reply sycl/include/CL/sycl/handler.hpp]: Fixed.

**Comment by bader** [comment]
@romanovvlad, please, resolve merge conflicts.

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 2]
```diff
@@ -17,10 +17,12 @@
 
 #include <assert.h>
 #include <cstdlib>
+#include <exception>
```
Comment (alexeyvoronov) [inline sycl/source/detail/program_manager/program_manager.cpp]: Why do you need include it?

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 3]
```diff
@@ -87,12 +89,19 @@ RT::PiProgram ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M,
...
-    Program = loadProgram(M, Context, &Img);
-    build(Program, Img->BuildOptions);
-  }
+  auto It = CachedPrograms.find(M);
+  if (It != CachedPrograms.end())
+    return It->second;
+
+  DeviceImage *Img = nullptr;
+  using PiProgramT = std::remove_pointer<RT::PiProgram>::type;
+  std::unique_ptr<PiProgramT, decltype(RT::piProgramRelease)> ProgramManaged(
```
Comment (alexeyvoronov) [inline sycl/source/detail/program_manager/program_manager.cpp]: use SYCL version unique_ptr_class

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 4]
```diff
@@ -87,12 +89,19 @@ RT::PiProgram ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M,
...
-    DeviceImage *Img = nullptr;
-    Program = loadProgram(M, Context, &Img);
-    build(Program, Img->BuildOptions);
-  }
+  auto It = CachedPrograms.find(M);
+  if (It != CachedPrograms.end())
+    return It->second;
+
+  DeviceImage *Img = nullptr;
+  using PiProgramT = std::remove_pointer<RT::PiProgram>::type;
```
Comment (alexeyvoronov) [inline sycl/source/detail/program_manager/program_manager.cpp]:
```suggestion
  using PiProgramT = remove_pointer_t<RT::PiProgram>;
```
from .../detail/type_traits.hpp

File: sycl/source/detail/program_manager/program_manager.cpp (hunk 2 above)
Comment (romanovvlad) [reply sycl/source/detail/program_manager/program_manager.cpp]: Fixed.

File: sycl/source/detail/program_manager/program_manager.cpp (hunk 3 above)
Comment (romanovvlad) [reply sycl/source/detail/program_manager/program_manager.cpp]: Done.

File: sycl/source/detail/program_manager/program_manager.cpp (hunk 4 above)
Comment (romanovvlad) [reply sycl/source/detail/program_manager/program_manager.cpp]: Done.

Comment (alexeyvoronov) [reply sycl/source/detail/program_manager/program_manager.cpp]: Please include the file that contains remove_pointer_t explicitly.

**Pull request merged**